Rakefile

```diff
@@ -1,25 +1,42 @@
-def xcpretty(cmd)
-  if `which xcpretty` && $?.success?
+def xcpretty(cmd, name)
+  if ENV['CI']
+    sh "set -o pipefail && #{cmd} | tee \"#{ENV['CIRCLE_ARTIFACTS']}/#{name}_raw.log\" | xcpretty --color --report junit --output \"#{ENV['CIRCLE_TEST_REPORTS']}/xcode/#{name}.xml\""
+  elsif `which xcpretty` && $?.success?
     sh "set -o pipefail && #{cmd} | xcpretty -c"
   else
     sh cmd
   end
 end
 
-task :build_for_testing do
-  sh "swift build"
-  xcpretty "xcodebuild -workspace StencilSwiftKit.xcworkspace -scheme Tests build-for-testing"
+def plain(cmd, name)
+  if ENV['CI']
+    sh "set -o pipefail && #{cmd} | tee \"#{ENV['CIRCLE_ARTIFACTS']}/#{name}_raw.log\""
+  else
+    sh cmd
+  end
+end
+
+task :spm_build do
+  plain("swift build", "spm_build")
+end
+
+task :xcode_build do
+  xcpretty("xcodebuild -workspace StencilSwiftKit.xcworkspace -scheme Tests build-for-testing", "xcode_build")
+end
+
+desc 'Run Xcode Unit Tests'
+task :xcode_test => :xcode_build do
+  xcpretty("xcodebuild -workspace StencilSwiftKit.xcworkspace -scheme Tests test-without-building", "xcode_test")
 end
 
-desc 'Run Unit Tests'
-task :test => :build_for_testing do
-  sh "swift test"
-  xcpretty "xcodebuild -workspace StencilSwiftKit.xcworkspace -scheme Tests test-without-building"
+desc 'Run SPM Unit Tests'
+task :spm_test => :spm_build do
```

Comment:
Tip: you can use take namespaces to group relative tasks together.

A `namespace 'spm' do … end` and a `namespace 'xcodebuild' do … end` would be nicer to read, and then you can name your tasks just `build` and `test` and refer to them as `rake spm:test` etc.